pkg/system/context.go

```diff
@@ -19,19 +19,14 @@ func WithSignalShutdown(parent context.Context) (
 
 	ctx, cancel := context.WithCancel(parent)
 	go func(ch chan os.Signal, cancel context.CancelFunc) {
-		<-ch
-		cancel()
+		select {
+		case <-ch:
+			cancel()
+
+		// Clean-up goroutine if the context is cancelled:
+		case <-ctx.Done():
+		}
 	}(ch, cancel)
 
 	return ctx, cancel
 }
-
-// OnCancel calls the given callback function when the provided context is
-// cancelled. Can be used to register clean-up callbacks for long-running
-// system contexts.
-func OnCancel(ctx context.Context, fn func()) {
```

Comment:
why deleted?

Reply:
i don't think i deleted? i'll add this back in just in case

Reply:
Was this a change that Guy made?